src/appleseed/renderer/modeling/bsdf/glassbsdf.cpp

```diff
@@ -228,70 +230,183 @@ namespace
                 values->m_anisotropy,
                 alpha_x,
                 alpha_y);
-            const float gamma = highlight_falloff_to_gama(values->m_highlight_falloff);
+
+            sampling_context.split_in_place(4, 1);
+            const Vector4f s = sampling_context.next2<Vector4f>();
 
             const Vector3f wo = basis.transform_to_local(sample.m_outgoing.get_value());
+            Vector3f wi;
+
+            bool is_refraction;
 
+            switch (m_mdf_type)
+            {
+                case GGX:
+                {
+                    const GGXMDF mdf;
+                    is_refraction = do_sample(
+                        mdf,
+                        s,
+                        adjoint,
+                        basis,
+                        alpha_x,
+                        alpha_y,
+                        0.0f,
+                        values->m_precomputed.m_reflection_color,
```

Comment:
Would it make sense to simply pass `values->m_precomputed` and `sample` instead of the individual fields? Right now the method takes 16 samples, which seems a bit excessive (and potentially inefficient since it almost certainly won't all fit in registers).

Reply:
I want to use do_sample(), which is a template now, outside of the normal rendering loop, 
without InputValues or BSDF samples and with floats, not Spectrums.
I plan to use it to compute EC tables. It could also be useful for random walk SSS.

Reply:
I want to use do_sample(), which is a template now, outside of the normal rendering loop,
without InputValues or BSDF samples and with floats, not Spectrums.
I plan to use it to compute EC tables. It could also be useful for random walk SSS.

Reply:
Alright.